packages/expo-router/src/types.ts

```diff
@@ -16,8 +16,10 @@ export interface RequireContext {
 export type PickPartial<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
 
 export type Router = {
-  /** Navigate to the provided href. */
+  /** Navigate to the provided href using a push operation if possible. */
   push: (href: Href) => void;
+  /** Navigate to the provided href. */
```

Comment:
Make this comment more explicit, indicating that it will always push, and that `pushOrPop` may be desired.